src/Pos/Block/Logic/Internal.hs

```diff
@@ -167,6 +168,7 @@ applyBlocksUnsafeDo blunds pModifier = do
         , slogBatch
         ]
     sscNormalize
+    dlgNormalize @ssc
```

Comment:
I don't like that it's done here. This normalization traverses all blocks from the epoch and might be quite expensive. And it should be done only after rollback. So I propose to move it to `rollbackBlocksUnsafe` and write few comments about all this stuff.

Also they way normalization is done after block application looks correct but is currently fragile too. If we add more rules, it might become broken. I see that it's not so easy to make it in the same way as in other components, so I don't insist on changing it.